### PR TITLE
Feature: add encryption and logging unit tests

### PR DESCRIPTION
**Pull request by MurashovVen**
- Adding `encryption` unit tests.
- Adding `logging` unit tests.
- Minor fixes:
bug fixing in `encryption/handler.go`;
code refactoring;
- Adding mocks.

### CODE REVIEW

**Review by Sriep — COMMENTED** [review]
In general I recommend duplicating tests to help expose race errors. I think this is particularly important for http.request handlers. 
There might be a few exceptions, like `InitLogging`, when the test function is only called once per execution.

File: code/go/0chain.net/core/logging/logger_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			name: "Test_InitLogging_testing_mode_OK",
+			args: args{mode: "testing"},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			InitLogging(tt.args.mode)
```
[the remark this answers is not in the document]
Comment (MurashovVen) [reply code/go/0chain.net/core/logging/logger_test.go:26]: Removed.

File: code/go/0chain.net/core/encryption/handler_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+				buf := bytes.NewBuffer(nil)
+				_, err := buf.WriteString(data)
+				require.NoError(t, err)
+				r := httptest.NewRequest(http.MethodGet, "/", buf)
+
+				return args{w: httptest.NewRecorder(), r: r}
+			}(),
+			wantW: w,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (MurashovVen) [reply code/go/0chain.net/core/encryption/handler_test.go:64]: Duplicated.

File: code/go/0chain.net/core/encryption/handler_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+				"signature": sign,
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			got, err := SignHandler(tt.args.ctx, tt.args.r)
```
[the remark this answers is not in the document]
Comment (MurashovVen) [reply code/go/0chain.net/core/encryption/handler_test.go:211]: Duplicated.

File: code/go/0chain.net/core/logging/handler_test.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+			}(),
+			want: w,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			N2NLogWriter(tt.args.w, tt.args.r)
```
[the remark this answers is not in the document]
Comment (MurashovVen) [reply code/go/0chain.net/core/logging/handler_test.go:115]: Duplicated.

File: code/go/0chain.net/core/logging/handler_test.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+			}(),
+			want: w,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			LogWriter(tt.args.w, tt.args.r)
```
[the remark this answers is not in the document]
Comment (MurashovVen) [reply code/go/0chain.net/core/logging/handler_test.go:58]: Duplicated.

**Review by Sriep — APPROVED** [review]

**Pull request merged**